src/pages/Staking/index.tsx

```diff
@@ -211,6 +211,7 @@ const Staking = (): JSX.Element => {
 
   // Estimated KINT Rewards & APY
   const monetaryLockingAmount = newMonetaryAmount(lockingAmount, GOVERNANCE_TOKEN, true);
+  const blockLockTimeExtension = convertWeeksToBlockNumbers(Number(lockTime));
```

Comment:
I think this is an issue since it causes a lot of rerenders when I check my console log. For example, updating the weeks once causes 10 - 20 rerenders which degrades performance of the app quite a bit.

Reply:
hi @nud3l this function is being called every render, so `blockLockTimeExtension` is being reset unnecessarily - the easiest fix here is to move it to an instance of `useState` and then use `useEffect` to only change it when the value of `lockTime` changes. I'll push up a fix.